### DISCUSSION TITLE
Why would this piece of WMI code fail to run if a console app is packaged and how to solve it?

### DISCUSSION THREAD

**Discussion by bogdan-patraucean** [dotnet/runtime#100203] [Q&A]
```
using System.Management;

namespace ConsoleApp1
{
    internal class Program
    {
        static void Main(string[] args)
        {
            ManagementScope scope = new(@"root\DEFAULT");
            ManagementPath path = new("SystemRestore");
            ManagementClass mClass = new(scope, path, new ObjectGetOptions());
            var restorePoints = mClass.GetInstances().Count;

            Console.WriteLine(restorePoints);
        }
    }
}
```
All I want to do is to count the system restore points, but this seems to be more difficult than I thought.

I have a WinAppSDK packaged app, and first I thought the issue was related to WinAppSDK, but as pointed by @DarranRowe in this [discussion](https://github.com/microsoft/WindowsAppSDK/discussions/4286#discussion-6413262), seems it could be a problem with .NET runtime.

So, the console app, will run just fine and the code will execute **elevated**, but ones the console app is packaged using a C# application packaging project, it fails with the same error it fails in the WinAppSDK app:

`System.Runtime.InteropServices.COMException (0x80040154)`

What could be the cause of this error in a packaged app? Should I open an issue?

**Comment by huoyaoyuan** [comment]
There's probably nothing to do with .NET runtime. The packaged app process is not granted for enough permission.

**Comment by bogdan-patraucean** [comment]
Up
